modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java

```diff
@@ -986,6 +986,165 @@ public void testDeleteObjectEntryAccountRestriction() throws Exception {
+				objectDefinition2.getObjectDefinitionId(), 0,
+				ObjectRelationshipConstants.DELETION_TYPE_PREVENT,
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				"oneToManyRelationship",
+				ObjectRelationshipConstants.TYPE_ONE_TO_MANY);
+
+		_objectDefinitionLocalService.publishSystemObjectDefinition(
+			_adminUser.getUserId(), objectDefinition2.getObjectDefinitionId());
+
+		ObjectEntry objectEntry1 = _objectEntryManager.addObjectEntry(
+			_simpleDTOConverterContext, objectDefinition1,
+			new ObjectEntry() {
+				{
+					properties = HashMapBuilder.<String, Object>put(
+						"textObjectFieldName", RandomTestUtil.randomString()
+					).build();
+				}
+			},
+			ObjectDefinitionConstants.SCOPE_COMPANY);
+
+		ObjectEntry objectEntry2 = _objectEntryManager.addObjectEntry(
+			_simpleDTOConverterContext, objectDefinition2,
+			new ObjectEntry() {
+				{
+					properties = HashMapBuilder.<String, Object>put(
+						"r_oneToManyRelationship_" +
+							objectDefinition1.getPKObjectFieldName(),
+						objectEntry1.getId()
+					).build();
+				}
+			},
+			ObjectDefinitionConstants.SCOPE_COMPANY);
+
+		Role role = _roleLocalService.addRole(
+			_adminUser.getUserId(), Role.class.getName(), 0,
+			RandomTestUtil.randomString(),
+			LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+			Collections.emptyMap(), RoleConstants.TYPE_REGULAR,
+			StringPool.BLANK, null);
+
+		_resourcePermissionLocalService.setResourcePermissions(
+			_companyId, objectDefinition1.getClassName(),
+			ResourceConstants.SCOPE_COMPANY, String.valueOf(_companyId),
+			role.getRoleId(),
+			new String[] {
+				ActionKeys.DELETE, ActionKeys.PERMISSIONS, ActionKeys.UPDATE,
+				ActionKeys.VIEW
+			});
+
+		_user = _addUser();
+
+		_userLocalService.addRoleUser(role.getRoleId(), _user);
+
+		try {
+			_objectEntryManager.deleteObjectEntry(
+				objectEntry1.getExternalReferenceCode(), _companyId,
+				objectDefinition1, null);
+
+			Assert.fail();
+		}
+		catch (PortalException portalException) {
+			Assert.assertEquals(
+				StringBundler.concat(
+					"Object relationship ",
+					objectRelationship.getObjectRelationshipId(),
+					" does not allow deletes"),
+				portalException.getMessage());
+		}
+
+		objectRelationship =
+			_objectRelationshipLocalService.updateObjectRelationship(
+				objectRelationship.getObjectRelationshipId(), 0,
+				ObjectRelationshipConstants.DELETION_TYPE_CASCADE,
+				objectRelationship.getLabelMap());
+
+		try {
+			_objectEntryManager.deleteObjectEntry(
+				objectEntry1.getExternalReferenceCode(), _companyId,
+				objectDefinition1, null);
+
+			Assert.fail();
+		}
+		catch (ObjectRelationshipDeletionTypeException
+					objectRelationshipDeletionTypeException) {
+
+			String message =
+				objectRelationshipDeletionTypeException.getMessage();
+
+			Assert.assertTrue(
+				message.contains(
+					"User " + _user.getUserId() +
+						" must have DELETE permission for"));
+		}
```

Comment:
@murilostodolni Not sure I understand the expected behavior here. Based on the description in [LPS-140522](https://issues.liferay.com/browse/LPS-140522):

> Cascade: Entry is deleted and all its relations entries on Object B are deleted too

Could you please add some comments to make it more clear?

Reply:
@carolmariaabb Luiz updated the expect behavior in this [comment](https://issues.liferay.com/browse/LPS-140522#:~:text=from%20Object%20A%3A-,Prevent%3A%20Entry%20cannot%20be%20deleted,delete%20the%20entry%20doesn%27t%20matter%20the%20permission%20in%20the%20related%20object,-cc%3A%20Samuel). I solved the bug and create the test with this behavior.